src/Raven.Server/Config/Categories/SecurityConfiguration.cs

```diff
@@ -1,63 +1,143 @@
+using System;
 using System.ComponentModel;
+using System.Linq;
+using System.Net;
 using Raven.Server.Config.Attributes;
+using Raven.Server.Config.Settings;
+using Raven.Server.Utils;
 
 namespace Raven.Server.Config.Categories
 {
     public class SecurityConfiguration : ConfigurationCategory
     {
-        [Description("Enables or disables the authentication for the server.")]
-        [DefaultValue(false)]
-        [ConfigurationEntry("Security.Authentication.Enabled")]
-        public bool AuthenticationEnabled { get; set; }
-        
-        [Description("If server is binded to non-localhost e.g. 'http://0.0.0.0:8080' then we require authentication to be enabled unless this setting is set to false.")]
-        [DefaultValue(true)]
-        [ConfigurationEntry("Security.Authentication.RequiredForPublicNetworks")]
-        public bool AuthenticationRequiredForPublicNetworks { get; set; }
-
-        [Description("The path of the .pfx certificate file. If specified, RavenDB will use HTTPS / SSL for all network activities. You can use the ~/ prefix to refer to RavenDB's base directory.")]
+        [Description(
+            "The path of the .pfx certificate file. If specified, RavenDB will use HTTPS / SSL for all network activities. You can use the ~/ prefix to refer to RavenDB's base directory.")]
         [DefaultValue(null)]
-        [ConfigurationEntry("Security.Certificate:Path")]
+        [ConfigurationEntry("Security.Certificate.Path")]
```

Comment:
No, this is wrong, we explicitly ask for ':' for a reasons.
This allows us to define configuration like:

```
"Security.Certificate": {
  "Path": "...",
  "Password": "..."
}
```